pkg/xds/server/components.go

```diff
@@ -225,10 +244,22 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler, ingressRec
 					return ingressReconciler.Reconcile(envoyCtx, &proxy)
 				}
 
+				snapshotHash, err := meshSnapshotCache.GetHash(ctx, dataplane.GetMeta().GetMesh())
+				if err != nil {
+					return err
+				}
+				if prevHash != "" && snapshotHash == prevHash {
+					log.V(1).Info("snapshot hashes are equal, no need to reconcile", "prev", prevHash)
```

Comment:
I'm slightly concerned about logging this even on debug. This can be a really large log.